### PR TITLE
Build all of our public externals in travis

### PR DESCRIPTION
**Pull request by rdeits**

### CODE REVIEW

**Comment by RussTedrake** [comment]
Can u merge master into this one please?  (Sorry)

**Comment by rdeits** [comment]
conflicts resolved (and should be prevented in the future by #1270)

**Comment by rdeits** [comment]
Yikes. drake005 is getting segfaults when starting matlab during the build.

**Comment by RussTedrake** [comment]
should be resolved now (just merged my fixes)

File: .travis.yml [hunk 1]
```diff
@@ -22,9 +27,12 @@ compiler:
   - gcc
   - clang
 
+python:
+  - '2.7'
+
 env:
   global:
-    - CMAKE_FLAGS="-DWITH_SPOTLESS:BOOL=OFF"
+    - CMAKE_FLAGS="-DWITH_SPOTLESS:BOOL=OFF -DWITH_LIBBOT:BOOL=ON -DWITH_GTK:BOOL=ON -DWITH_DIRECTOR:BOOL=ON -DWITH_IRIS:BOOL=ON-DWITH_OCTOMAP:BOOL=ON -DWITH_MOSEK:BOOL=ON -DWITH_AVL:BOOL=ON -DWITH_XFOIL:BOOL=ON"
```
Comment (RussTedrake) [inline .travis.yml]: why not -DWITH_ALL_SUPPORTED_EXTERNALS=ON ?  it's more likely to stay up to date.

Comment (rdeits) [inline .travis.yml]: Some of the supported externals require matlab but don't check whether it
exists yet.

On Wed, Aug 19, 2015 at 5:02 PM, Russ Tedrake <notifications@github.com>
wrote:

> In .travis.yml
> <https://github.com/RobotLocomotion/drake/pull/1254#discussion_r37467602>:
>
> >  env:
> >    global:
> > -    - CMAKE_FLAGS="-DWITH_SPOTLESS:BOOL=OFF"
> > +    - CMAKE_FLAGS="-DWITH_SPOTLESS:BOOL=OFF -DWITH_LIBBOT:BOOL=ON -DWITH_GTK:BOOL=ON -DWITH_DIRECTOR:BOOL=ON -DWITH_IRIS:BOOL=ON-DWITH_OCTOMAP:BOOL=ON -DWITH_MOSEK:BOOL=ON -DWITH_AVL:BOOL=ON -DWITH_XFOIL:BOOL=ON"
>
> why not -DWITH_ALL_SUPPORTED_EXTERNALS=ON ? it's more likely to stay up to
> date.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/1254/files#r37467602>.
>



-- 
Robin L. H. Deits
Robot Locomotion Group @CSAIL
Massachusetts Institute of Technology

Comment (RussTedrake) [inline .travis.yml]: ah.  got it.

**Comment by RussTedrake** [comment]
i think the test failures are still for the updated iris, correct?  
does that mean this pull request subsumes #1223 ?  (can we close the other one)?

**Comment by rdeits** [comment]
Interesting. drake005 is failing to open the `libmosek64.7.1.dylib`, but I can verify that it's been installed to build/lib. @RussTedrake how do the build servers set their (DY)LD_LIBRARY_PATH?

**Comment by RussTedrake** [comment]
hah.  i try not to force them to do that.  check out the bullet makefile — i install the full path to the library into the rpath (justified by the fact that these are local installs)

**Comment by rdeits** [comment]
@RussTedrake can you re-trigger the build servers for this one now that they have swig?

**Comment by RussTedrake** [comment]
just close and reopen the pull.

**Comment by rdeits** [comment]
So close...just need to fix the library install path for mosek on linux.

**Comment by rdeits** [comment]
Actually, I'm confused. drake006 has one failing unit tests with an invalid mex file error on the IRIS mexfile, because it can't find libmosek. But that mex file is loaded by several other tests which didn't fail, and I can load it on my system (with no LD_LIBRARY_PATH configuration).

**Comment by RussTedrake** [comment]
not sure.  did you log on to drake006 to try an reproduce?

**Comment by rdeits** [comment]
Ok, I can reproduce it now (it wasn't happening before because I have *tons* of copies of the mosek libraries on my various computers). 

The problem is that I link against libmosek by using the pods_use_pkg_config_packages macro. The mosek.pc file correctly sets the paths to find the mosek shared libraries, so compilation is always fine. On OSX, I'm using install_name_tool to change the id of the mosek shared library to its full path, which results in the correct location of the library being used at runtime. But install_name_tool isn't a thing on linux, and I'm not sure what the canonical way of doing this is.

**Comment by rdeits** [comment]
I think I got it this time.

File: .travis.yml (hunk 1 above)
Comment (patmarion) [inline .travis.yml]: looks like there is a typo?  Missing a space here:

-DWITH_IRIS:BOOL=ON-DWITH_OCTOMAP:BOOL=ON

Comment (rdeits) [inline .travis.yml]: thanks, just pushed a fix

**Comment by rdeits** [comment]
@RussTedrake this currently points the mosek pod to my branch: https://github.com/rdeits/mosek-pod/tree/rd-install-lib Note that that branch includes the logic from #1266 which we still haven't decided if we want.

**Comment by rdeits** [comment]
Uh oh. I think the travis machine is running out of RAM (or possibly timing out) while trying to build drakeGeometryUtil in debug mode.

**Comment by rdeits** [comment]
reopening to give drake003 another chance

**Comment by rdeits** [comment]
The only failure on this is the Travis server running out of resources while compiling drakeGeometryUtil again. None of that code changed on this pull request, and it didn't happen on a prior, identical, build. I suspect that we're using close to the maximum amount of the RAM and swap available, and it's failing with probability based on load. 

I think this is probably ready to merge.

**Comment by RussTedrake** [comment]
that's going to be very annoying (trying to decide why travis fails each time), and it's also worrisome that compiling has gotten so hard.  I'll merge this, but i think we need to resolve it...

**Pull request merged**